src/web_accessible/checkPageInteraction.js

```diff
@@ -0,0 +1,97 @@
+          backdrop-filter: blur(5px);
+        }
+        #swph__dialog {
+          max-width: 50vh;
+          padding: 0.75em;
+          border-radius: 8px;
+          user-select: none;
+          text-align: center;
+        }
+        #swph__dialog img {
+          height: 1.5em;
+        }
+        #swph__dialog p {
+          line-height: 1.25em;
+          padding: 0.25em;
+        }
+        #swph__dialog-title {
+          display: flex;
+          flex-direction: row;
+          align-items: center;
+          justify-content: center;
+          font-weight: bold;
+        }
+        #swph__dialog-text {
+          font-size: 0.95em;
+        }
+        #swph__dialog-subtext {
+          opacity: 0.65;
+          font-size: 0.8em;
+        }
+			`.replace(/^\s+/gm, '');
+      dialog = document.createElement('dialog');
+      dialog.id = 'swph__dialog';
+      dialog.innerHTML = `
+        <p id="swph__dialog-title"><img src="${chrome.runtime.getURL('web_accessible/logo.svg')}">&nbsp;<span>${chrome.i18n.getMessage('application_title')}</span></p>
+        <p id="swph__dialog-text">${chrome.i18n.getMessage('interact_dialog_text')}</p>
+        <p id="swph__dialog-subtext">${chrome.i18n.getMessage('interact_dialog_subtext')}</p>
+      `.replace(/^\s+/gm, '');
+      [dialogStyle, dialog].forEach((e) => document.body.insertAdjacentElement('beforeend', e));
+      // Stop 'Esc' key from closing modal by default - doesn't count as user interaction.
+      // https://www.zdnet.com/article/google-changes-how-the-escape-key-is-handled-in-chrome-to-fight-popup-ads/
+      dialog.addEventListener('cancel', (ev) => ev.preventDefault());
+
+      function reactToUserInteraction(ev) {
+        if (!window.navigator.userActivation.hasBeenActive) return;
+				if (dialog.dataset.swphLastCommand) successCallback(dialog.dataset.swphLastCommand);
+        document.querySelectorAll('[id*="swph__"').forEach((e) => e.remove());
+        ['keydown', 'pointerdown'].forEach((evType) => window.removeEventListener(evType, reactToUserInteraction));
+      }
+      ['keydown', 'pointerdown'].forEach((evType) => window.addEventListener(evType, reactToUserInteraction, { passive: true }));
+    }
+    dialog.close();
+
+    if (['play-pause', 'next', 'previous'].includes(command)) {
+      dialog.dataset.swphLastCommand = command;
+      // Only block commands that actually affect playback - 'previous' & 'next'
+      // don't always start playback but can sometimes throw user out of current
+      // queue if paused & skipping to an advert.
+      dialog.showModal();
+      // Send response to tell service worker to focus Spotify tab.
+      return 'requestUserInteraction';
+    } else {
+      delete dialog.dataset.swphLastCommand;
+      // Allow other commands but with unobtrusive visible warning, unless switching
+      // to search page - don't want to get in the way when the search input element
+      // receives focus automatically.
```

Comment:
I've realised if you do retain the non-modal dialog for certain situations, we also need to add a high positive z-index to the `dialogStyle` css. It would catch a rare case where the dialog element is added prior to the main page-body being visible.
(If you open / reload the spotify tab and then _immediately_ attempt hotkey playback the modal is always shown correctly above everything, but when it switches to a normal dialog it's hidden below the `#main` div).

Reply:
Just depends which way you'd prefer. I don't know if it's potentially more confusing to have the dialog disappear entirely or to have it occasionally presented in a non-blocking manner after the user has seen but not acted upon the modal version.

Reply:
If I understand correctly this discussion, I'd just drop the concept of "non modal / 'priming' dialog" and go with the simple approach of requiring attention (showing the modal) when absolutely necessary (playback stuff). Sounds good?

Reply:
Yep I've implemented option 1. (As we still need to delete the dataset prop to avoid unexpectedly dispatching play / next / previous after the user has issued a different command before interacting).